packages/webapp/src/containers/SensorReadings/index.jsx

```diff
@@ -1,78 +1,52 @@
 import { React, useEffect, useState } from 'react';
 import { useTranslation } from 'react-i18next';
-import { useSelector } from 'react-redux';
+import { useSelector, useDispatch } from 'react-redux';
 import SensorReadingsLineChart from '../SensorReadingsLineChart';
-import {
-  CURRENT_DATE_TIME,
-  SOIL_WATER_POTENTIAL,
-  SOIL_WATER_CONTENT,
-  TEMPERATURE,
-} from './constants';
 import PageTitle from '../../components/PageTitle/v2';
 import RouterTab from '../../components/RouterTab';
-import { bulkSensorsReadingsSliceSelector } from '../bulkSensorReadingsSlice';
 import { sensorsSelector } from '../sensorSlice';
-import { ambientTemperature, soilWaterPotential } from '../../util/convert-units/unit';
-import { getUnitOptionMap } from '../../util/convert-units/getUnitOptionMap';
-import { measurementSelector } from '../../containers/userFarmSlice';
-import styles from './styles.module.scss';
-import { Semibold } from '../../components/Typography';
 import { sensorReadingTypesByLocationSelector } from '../../containers/sensorReadingTypesSlice';
+import { getSensorsReadings } from '../SensorReadings/saga';
+import { bulkSensorsReadingsSliceSelector } from '../bulkSensorReadingsSlice';
 
 function SensorReadings({ history, match }) {
   const { t } = useTranslation();
-  const { location_id = '' } = match?.params;
-  const sensorInfo = useSelector(sensorsSelector(location_id));
-  const {
-    latestMinTemperature = '',
-    latestMaxTemperature = '',
-    nearestStationName = '',
-    lastUpdatedReadingsTime = '',
-    predictedXAxisLabel = '',
-    xAxisLabel = {},
-  } = useSelector(bulkSensorsReadingsSliceSelector);
+  const dispatch = useDispatch();
+  const { location_id } = match?.params;
   const [readingTypes, setReadingTypes] = useState([]);
-  const unitSystem = useSelector(measurementSelector);
+  const [locationData, setLocationData] = useState();
+
+  const sensorInfo = useSelector(sensorsSelector(location_id));
   const reading_types = useSelector(sensorReadingTypesByLocationSelector(location_id));
```

Comment:
This might fall under the "I tried to to change everything rule" haha . 

But I like this idea so I just tried it. Problem seems to be that dispatch needs it -- so the dispatch fails when I replace the state with just the selector.